zds/member/views.py

```diff
@@ -481,18 +479,16 @@ def unregister(request):
         topic.participants.remove(current)
         topic.save()
     Topic.objects.filter(author=current).update(author=anonymous)
-    # Before deleting gallery let's summurize what we deleted
-    # - unpublished tutorials with only the unregistering member as an author
-    # - unpublished articles with only the unregistering member as an author
-    # - all category associated with those entites (have a look on article.delete_entity_and_tree
-    # and tutorial.delete_entity_and_tree
-    # So concerning galleries, we just have for us
-    # - "personnal galleries" with only one owner (unregistering user)
-    # - "personnal galleries" with more than one owner
-    # so we will just delete the unretistering user ownership and give it to anonymous in the only case
-    # he was alone so that gallery is not lost
+    # All contents with only the unregistering member as an author will
+    # be deleted just before the User object (with a pre_delete receiver).
+    # So concerning galleries, we just have for us:
+    # - "personal galleries" with only one owner (unregistering user);
+    # - "personal galleries" with more than one owner.
+    # So we will just remove the unregistering user's ownership and
+    # give it to anonymous in the only case they were alone so that
+    # gallery is not lost.
```

Comment:
gallery 1 belongs to user 1
user1 want's to leave
now gallery1 belongs to anonymous user
user1 is deleted